### DISCUSSION TITLE
Is line highlighting supported?

### DISCUSSION THREAD

**Discussion by johnnymetz** [adityatelange/hugo-PaperMod#525] [Q&A] [answered]
In other themes, I can highlight a specific line of code (using two backticks below instead of three for display purposes):

```
```python {hl_lines=[2]}
def hello(name):
    print(f"Hello {name}!")
``
```

Which renders something like this:

![Screen Shot 2021-08-13 at 12 28 06 AM](https://user-images.githubusercontent.com/20693145/129320613-6c5a710c-2ef2-4dba-945f-19fef519b9f5.png)

However, using this theme I don't see any highlight on line 2:

![Screen Shot 2021-08-13 at 12 27 28 AM](https://user-images.githubusercontent.com/20693145/129320635-88a39291-f139-42d5-a72f-927357d3d5c9.png)

Is this a bug or is this not supported?

**Comment by xiaogit00** [comment]
facing the same problem

**Comment by noga-dev** [comment, marked as answer]
Remove this rule to restore line highlighting functionality:

https://github.com/adityatelange/hugo-PaperMod/blob/4e9b04784bac6bbaabecdffc9fe6e513787b66d9/assets/css/common/post-single.css#L196-L198

**Reply by johnnymetz** [reply]
This works perfectly. Here are images with and without line numbers:

<img width="910" alt="Screen Shot 2021-09-30 at 12 36 17 AM" src="https://user-images.githubusercontent.com/20693145/135408297-e2eb4efd-1555-4225-b300-86a646bafa2a.png">

<img width="908" alt="Screen Shot 2021-09-30 at 12 34 55 AM" src="https://user-images.githubusercontent.com/20693145/135408313-b97d5a5b-73f9-403b-a40d-eba6a74bd7f4.png">

**Reply by johnnymetz** [reply]
Created a PR to fix this here: https://github.com/adityatelange/hugo-PaperMod/pull/575

**Comment by JoJoJotarou** [comment]
Now, in master, I got this problem.😥

![111](https://img1.imgtp.com/2022/05/18/fotQoYnk.png)

**Reply by JoJoJotarou** [reply]
I found a solution. In mine old config.yml:

```yml
markup:
    highlight:
        # anchorLineNos: true
        codeFences: true
        guessSyntax: true
        lineNos: true
```
this
`````html
```html
```
`````
is same as this.
`````html
```html {linenos=table}
```
`````

**Means the codeblock is rendered as `<table>` by default.**

By https://gohugo.io/functions/highlight/#options
> Instead of specifying both lineNos and lineNumbersInTable, you can use the following shorthand notation:
>
> lineNos=inline
> equivalent to lineNos=true and lineNumbersInTable=false
> lineNos=table
> equivalent to lineNos=true and lineNumbersInTable=true

So add `lineNumbersInTable=false` in config.yml to tell Hugo not use `<table>` rendering codeblock.

```yml
markup:
    highlight:
        lineNumbersInTable: false
```
now, this
`````html
```html
```
`````
is same as this.
`````html
```html {linenos=inline}
```
`````

I am sorry that there may be a little bit inconsistent with the theme, I hope to help who have the same problem, so I will not delete the reply, thk.🎇

hugo: v0.95
PaperMod(commit Id): [cb9be55c78efadee651f3a9f5e9b719f28a6d290](https://github.com/adityatelange/hugo-PaperMod/commit/cb9be55c78efadee651f3a9f5e9b719f28a6d290)
